flake-modules/dev/new-plugin.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+import os
+import re
+from argparse import ArgumentParser
+
+# Template for default.nix
+# TODO: conditionally include parts of the template based on args
+default_nix_template = """{{ lib, ... }}:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {{
+  name = "{name}";
+  originalName = "{originalName}";
+  package = "{package}";
+
+  maintainers = [ lib.maintainers.YOUR_NAME ];
+
+  settingsOptions = {{
+    exampleOption = lib.nixvim.defaultNullOpts.mkBool false ''
+      Example option for the {name} plugin.
+    '';
+  }};
+}}
+"""
+
+# Template for test file
+test_nix_template = """{{
+  empty = {{
+    plugins.{name}.enable = true;
+  }};
+}}
+"""
+
+
+def kebab_case(input_string):
+    """
+    Convert a string to kebab-case.
+
+    Args:
+        input_string (str): The input string to convert.
+
+    Returns:
+        str: The converted kebab-case string.
+    """
+    input_string = re.sub(r"(?<!^)(?=[A-Z])", "-", input_string)
+
+    return re.sub(r"[\W_]+", "-", input_string).lower().strip("nvim").strip("-")
```

Comment:
Can `strip()` accept a regex? If so we could leverage word boundaries to remove some edge-cases. If not, no big deal.

 Maybe also catch the `-` in one pass using `-*`, although that may be less readable than chaining multiple `.strip()` calls...

```suggestion
    return re.sub(r"[\W_]+", "-", input_string).lower().strip(r"\bnvim\b").strip("-")
```

Reply:
I went with chaining calls in case stripping `nvim` caused there to be a leftover `-`.

But, looks like `strip` doesn't accept a regex so we have to use `sub`